miso-web/src/main/webapp/scripts/list_attachment.js

```diff
@@ -177,6 +107,121 @@ ListTarget.attachment = (function() {
               Utils.page.pageReload();
             });
           });
+    },
+    showUploadDialog: function(entityType, entityId, sharedIds) {
+      var dialogArea = jQuery('#dialog');
+      dialogArea.empty();
+
+      var form = jQuery('<form id="uploadForm">');
+      form.append(jQuery('<p><input id="fileInput" type="file" name="file"></p>'));
+      var select = jQuery('<select id="attachmentCategory" type="text" name="attachmentCategory">');
+      [{
+        id: 0,
+        alias: 'Misc.'
+      }].concat(Constants.attachmentCategories).forEach(function(category) {
+        select.append(jQuery('<option value="' + category.id + '">' + category.alias + '</option>'));
+      });
+      var p = jQuery('<p>').append(jQuery('<label for="attachmentCategory">Category:</label>')).append(select);
+      form.append(p);
+      dialogArea.append(form);
+
+      var dialog = dialogArea.dialog({
+        autoOpen: true,
+        width: 500,
+        title: 'Attach File',
+        modal: true,
+        buttons: {
+          upload: {
+            id: 'upload',
+            text: 'Upload',
+            click: function() {
+              if (!jQuery('#fileInput').val()) {
+                alert('No file selected!');
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove this usage of alert(...). [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=javascript%3AS1442)